rn/Teacher/ios/Teacher/SpeedGrader/RubricAssessor.swift

```diff
@@ -92,78 +92,126 @@ struct RubricAssessor: View {
+                    .alignmentGuide(.leading, computeValue: leading)
+                    .alignmentGuide(.top, computeValue: top)
                 }
             }
-
-            let customGrade = (assignment.freeFormCriterionCommentsOnRubric || assessment?.rating_id == nil || assessment?.rating_id == "")
-                ? assessment?.points : nil
-            CircleToggle(isOn: Binding(get: { customGrade != nil }, set: { newValue in
-                if newValue {
-                    promptCustomGrade(criteria, assessment: assessment)
-                } else {
-                    assessments[criteria.id] = APIRubricAssessment(comments: assessment?.comments)
-                }
-            })) {
-                if let grade = customGrade {
-                    Text(grade)
-                } else {
-                    Image.addSolid
-                }
-            }
-                .accessibility(label: Text("Customize Grade"))
-                .alignmentGuide(.leading, computeValue: leading)
-                .alignmentGuide(.top, computeValue: top)
+            customGradeToggle(
+                criteria: criteria,
+                assessment: assessment,
+                leading: leading,
+                top: top
+            )
         }
-            .padding(.top, 8)
+        .padding(.top, 8)
 
         let showAdd = assignment.freeFormCriterionCommentsOnRubric && assessment?.comments?.isEmpty != false
         let showLong = criteria.longDesc.isEmpty == false
         if showAdd || showLong {
-            HStack(spacing: 6) {
-                if showAdd {
-                    Button(action: { withAnimation(.default) {
-                        comment = ""
-                        commentID = criteria.id
-                    } }, label: {
-                        Text("Add Comment")
-                            .font(.medium14).foregroundColor(.accentColor)
-                    })
-                        .identifier("SpeedGrader.Rubric.\(criteria.id).addCommentButton")
-                }
-                if showAdd, showLong {
-                    Text(verbatim: "•")
-                        .font(.regular12).foregroundColor(.textDark)
+            addButtons(
+                criteria: criteria,
+                assessment: assessment,
+                showAdd: showAdd,
+                showLong: showLong
+            )
+        }
+    }
+    }
+
+    private func customGradeToggle(
+        criteria: Rubric,
+        assessment: APIRubricAssessment?,
+        leading: @escaping (ViewDimensions) -> CGFloat,
+        top: @escaping (ViewDimensions) -> CGFloat
+    ) -> some View {
+        let customGrade = (
+            assignment.freeFormCriterionCommentsOnRubric ||
+                assessment?.rating_id == nil ||
+                assessment?.rating_id == ""
+        )
+            ? assessment?.points : nil
+
+        return CircleToggle(
+            isOn:
+            Binding(
```

Comment:
In such cases we could extract the Binding into a local variable (like `customGrade`) to increase readability.

Reply:
My reason behind defining it there is that it's not used anywhere else but in this local scope only.

Reply:
I extracted the binding into its own variable!